### PR TITLE
Update to use Poetry

### PR DESCRIPTION
**Pull request by lmarsden**

### CODE REVIEW

File: {{cookiecutter.project_repo}}/Makefile [hunk 1]
```diff
@@ -50,41 +50,44 @@ clean-test: ## remove test and coverage artifacts
 lint: checktypes checkstyle sast checklicenses ## run all checks
 
 checktypes: ## check types with mypy
-	mypy --ignore-missing-imports {{ cookiecutter.project_slug }} tests setup.py
+	mypy --ignore-missing-imports {{ cookiecutter.project_slug }} tests
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/Makefile:53]: This is something I missed on my original pass
```suggestion
	poetry run mypy --ignore-missing-imports {{ cookiecutter.project_slug }} tests
```

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+[tool.poetry]
+name = "{{ cookiecutter.project_slug }}"
+version = "{{ cookiecutter.version }}"
+authors = [
+    "UK Hydrographic Office <datascienceandengineering@ukho.gov.uk>"
+]
+description = "Python wrapper for the C implementation of the Generic Sensor Format library."
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:7]: I think there is a "{{ }}" variable for the description

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+pytest-cov = "^2.10.0"
+pytest-runner = "^5.2"
+pytest = "^5.4.3"
+tox-gh-actions = "^1.2.0"
+tox = "^3.15.2"
+toml = "^0.10.1"
+immutabledict = "^1.0.0"
+mypy = "^0.780"
+pytest-cases = "^1.16.0"
+dataclasses = {version = "^0.7", python = ">=3.6,<3.7"}
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:51]: can remove this line

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+liccheck = "^0.4.7"
+pytest-cov = "^2.10.0"
+pytest-runner = "^5.2"
+pytest = "^5.4.3"
+tox-gh-actions = "^1.2.0"
+tox = "^3.15.2"
+toml = "^0.10.1"
+immutabledict = "^1.0.0"
+mypy = "^0.780"
+pytest-cases = "^1.16.0"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:50]: can remove this line

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+flake8-isort = "^3.0.0"
+flake8 = "^3.8.3"
+liccheck = "^0.4.7"
+pytest-cov = "^2.10.0"
+pytest-runner = "^5.2"
+pytest = "^5.4.3"
+tox-gh-actions = "^1.2.0"
+tox = "^3.15.2"
+toml = "^0.10.1"
+immutabledict = "^1.0.0"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:48]: can remove this line

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+[tool.poetry.dependencies]
+python = "^3.6"
+
+[tool.poetry.dev-dependencies]
+assertpy = "^1.0"
+bandit = "^1.6.2"
+black = "^19.10b0"
+bump2version = "^1.0.0"
+coverage = "^5.1"
+flake8-isort = "^3.0.0"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:39]: can remove this line and add an explicit dependency on isort

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+# Do not manually edit, use poetry add/remove (--dev) to manage dependencies
+[tool.poetry.dependencies]
+python = "^3.6"
+
+[tool.poetry.dev-dependencies]
+assertpy = "^1.0"
+bandit = "^1.6.2"
+black = "^19.10b0"
+bump2version = "^1.0.0"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:37]: can remove this line

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+tox-gh-actions = "^1.2.0"
+tox = "^3.15.2"
+toml = "^0.10.1"
+immutabledict = "^1.0.0"
+mypy = "^0.780"
+pytest-cases = "^1.16.0"
+dataclasses = {version = "^0.7", python = ">=3.6,<3.7"}
+
+[tool.pytest.ini_options]
+addopts = "--cov=gsfpy --cov-fail-under=95"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:54]:
```suggestion
addopts = "--cov={{ cookiecutter.project_slug }} --cov-fail-under=95"
```

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+[build-system]
+requires = ["poetry>=0.12"]
+build-backend = "poetry.masonry.api"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:58]: new line at end of file

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8"
+]
+
+homepage = "https://github.com/UKHO/{{ cookiecutter.project_slug }}"
+repository = "https://github.com/UKHO/{{ cookiecutter.project_slug }}"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:25]:
```suggestion
homepage = "https://github.com/UKHO/{{ cookiecutter.project_repo }}"
repository = "https://github.com/UKHO/{{ cookiecutter.project_repo }}"
```

File: {{cookiecutter.project_repo}}/pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,58 @@
+[tool.poetry]
+name = "{{ cookiecutter.project_slug }}"
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/pyproject.toml:2]:
```suggestion
name = "{{ cookiecutter.project_repo }}"
```

File: {{cookiecutter.project_repo}}/.gitignore [hunk 12]
```diff
@@ -1,103 +1,39 @@
...
-
-# mypy
-.mypy_cache/
-
-# IDE settings
-.vscode/
-.idea/
+/releases/*
+pip-wheel-metadata
+/poetry.toml
```
Comment (eganjs) [inline {{cookiecutter.project_repo}}/.gitignore:39]: new line at end of file

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 10 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml:25]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 11 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml:2]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 2 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Fixed

File: {{cookiecutter.project_repo}}/Makefile (hunk 1 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/Makefile]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 8 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml:50]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 9 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml:54]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 3 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 4 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 5 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Done

File: {{cookiecutter.project_repo}}/.gitignore (hunk 12 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/.gitignore:39]: Fixed

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 6 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Done

File: {{cookiecutter.project_repo}}/pyproject.toml (hunk 7 above)
Comment (lmarsden) [reply {{cookiecutter.project_repo}}/pyproject.toml]: Done

**Comment by lmarsden** [comment]
> Remove `{{cookiecutter.project_repo}}/requirements-dev.txt`

Not sure how that snuck back in. Now fixed.

**Pull request merged**